### PR TITLE
bump-formula-pr, utils/pypi: tweak log messages

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

More useful in combination with #8147

- `bump-formula-pr` - pass in `args.quiet?` to `update_python_resources!` instead of always passing in `true` for `silent`
- `utils/pypi`
  - Add parentheses that say that retrieving PyPI dependencies "may take awhile"
  - Add message about getting PyPI info for each dependency to give a better sense of what is happening behind the scenes

### CODE REVIEW

File: Library/Homebrew/utils/pypi.rb [hunk 1]
```diff
@@ -52,7 +52,7 @@ def update_python_resources!(formula, version = nil, print_only: false, silent:
       odie "\"#{formula.name}\" contains non-PyPI resources. Please update the resources manually."
     end
 
-    ohai "Retrieving PyPI dependencies for \"#{pypi_name}==#{version}\"" if !print_only && !silent
+    ohai "Retrieving PyPI dependencies for \"#{pypi_name}==#{version}\" (this may take awhile)" if !print_only && !silent
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/pypi.rb:55]:
```suggestion
    ohai "Retrieving PyPI dependencies for \"#{pypi_name}==#{version}\"..." if !print_only && !silent
```

**Comment by dtrodrigues** [comment]
One thing I've noticed is that it always prints `brew update-python-resources <formulaname>`, even for non-python formulae. It may be worth changing where that message is logged because even though the command technically is called every time, it's a little confusing to always see.

**Comment by SeekingMeaning** [comment]
Maybe we should skip `PyPI.update_python_resources!` if the formula doesn't depend on any version of `python`

It seems all formulae using `pythonhosted` require `python`: https://github.com/Homebrew/homebrew-core/search?q=pythonhosted+NOT+python

**Comment by Rylan12** [comment]
There are 16 formulae that use `files.pythonhosted.org` but not `depends_on "python`:

```console
$ grep -L 'depends_on "python' $(grep -l 'files.pythonhosted.org' *)
appscale-tools.rb
blastem.rb
bup.rb
cassandra.rb
cassandra@2.1.rb
cassandra@2.2.rb
gr-osmosdr.rb
mesos.rb
numpy@1.16.rb
offlineimap.rb
ooniprobe.rb
pypy.rb
pypy3.rb
python@3.7.rb
python@3.8.rb
volatility.rb
```

Most of the are because they have `depends_on :macos` for python 2.

There are still a few that don't declare a python dependency though and have resources:
- `gr-osmosdr` (although `python@3.8` is a recursive dependency)
- `pypy`
- `pypy3`
- `python@3.7`
- `python@3.8`

**Comment by SeekingMeaning** [comment]
> Most of them are because they have `depends_on :macos` for python 2.

Maybe Python should have something like Java's `depends_on java: "1.8"`, perhaps `depends_on python: "2"`? Also maybe we should do this for Ruby too as some formula (or at least they used to) depend on built-in macOS Ruby



> There are still a few that don't declare a python dependency though and have resources:
> 
> * `gr-osmosdr` (although `python@3.8` is a recursive dependency)
> * `pypy`
> * `pypy3`
> * `python@3.7`
> * `python@3.8`

We probably could just hardcode `pypy` and versioned `python` since they're obviously Python-related packages

**Comment by jonchang** [comment]
> Maybe Python should have something like Java's `depends_on java: "1.8"`, perhaps `depends_on python: "2"`? Also maybe we should do this for Ruby too as some formula (or at least they used to) depend on built-in macOS Ruby

We're trying to move away from Requirements in Homebrew/core. I think the approach of having a hardcoded list of Python formulae to check against for the purposes of updating Python resources is fine. We do that in `languages/python.rb` anyway.

**Comment by MikeMcQuaid** [comment]
> One thing I've noticed is that it always prints `brew update-python-resources <formulaname>`, even for non-python formulae. It may be worth changing where that message is logged because even though the command technically is called every time, it's a little confusing to always see.

How slow is this to run? If it's not slow: let's just not print that it's run unless something is changed in the files. This seems preferable to maintaining a allowlist/denylist just to avoid a message being printed.

File: Library/Homebrew/dev-cmd/bump-formula-pr.rb [hunk 2]
```diff
@@ -333,7 +333,7 @@ def bump_formula_pr
 
     ohai "brew update-python-resources #{formula.name}"
```
Comment (SeekingMeaning) [inline Library/Homebrew/dev-cmd/bump-formula-pr.rb:334]:
```suggestion
```

This line probably isn't needed anymore if we pass `silent: args.quiet?` to `PyPI.update_python_resources!` (whereas previously we always passed `silent: true`)

Example output without PyPI dependencies:

```console
$ brew bump-formula-pr helmfile --version 0.125.3 --dry-run --write --force
...
==> Downloading https://github.com/roboll/helmfile/archive/v0.125.3.tar.gz
==> replace /https:\/\/github\.com\/roboll\/helmfile\/archive\/v0\.125\.2\.tar\.gz/ with "https://github.com/roboll/helm
==> replace "72d3b5646459408b8096c94408ea5fdf72c335575b99e3c0072fe81320b98c6c" with "2ca6b039c72414eaf2ffe5848f61320f8ae
==> brew audit helmfile.rb
...
```

With PyPI dependencies:

```console
$ brew bump-formula-pr jc --version 1.13.4 --dry-run --write --force
...
==> Downloading https://files.pythonhosted.org/packages/e3/b9/7878a4f71c873c7d67f39615086f1c8315740534b25eddc1a4f75f3148
==> replace /https:\/\/files\.pythonhosted\.org\/packages\/2b\/1e\/179eea9186313bcff8dc3405ecd4615043ea615681519a9638d79
==> replace "b5ebb419b3b5d3cd95a166b4f156c7986235e983d5c6bf21aa9d57586e211f78" with "45480ac3d399f70b57d8cc97a6795ea875a
==> Retrieving PyPI dependencies for "jc==1.13.4"...
==> Getting PyPI info for "pygments==2.6.1"
==> Getting PyPI info for "ruamel-yaml==0.16.10"
==> Getting PyPI info for "ruamel-yaml-clib==0.2.0"
==> Getting PyPI info for "xmltodict==0.12.0"
==> Updating resource blocks
==> brew audit jc.rb
...
```

**Pull request merged**